### PR TITLE
Fix for FastSet#has always returning true in some situations.

### LINKED ISSUE #98: Bug in FastSet: `.contains(s)` is true for all single-character strings `s`
An empty, newly-created FastSet claims to contain all single-character strings.

### Repro:
```
$ npm install collections
```

And run the following test code:
```
var FastSet = require('collections/fast-set');
var assert = require('assert');

var s = new FastSet();
assert.equal(s.has('hello'), false); // succeeds
assert.equal(s.has('h'), false); // fails, and seems to fail for any single-character string

assert.deepEqual(s.buckets.getDefault('hello').toArray(), ['h', 'e', 'l', 'l', 'o']); // succeeds
```

### PR DESCRIPTION
**Pull request by vlad-alexandru-ionescu**
eg: key is a single letter string.
Fixes #98.

### CODE REVIEW

File: fast-set.js [hunk 1]
```diff
@@ -22,7 +22,9 @@ function FastSet(values, equals, hash, getDefault) {
     this.contentEquals = equals;
     this.contentHash = hash;
     this.getDefault = getDefault;
-    this.buckets = new this.Buckets(null, this.Bucket);
+    this.buckets = new this.Buckets(null, (function () {
+        return new this.Bucket();
+    }).bind(this));
```
Comment (kriskowal) [inline fast-set.js]: In V8, bind is slower than just capturing `this` as `self` in the parent scope and calling `new self.Bucket()`.

**Comment by kriskowal** [comment]
This is a great catch. getDefault gets called with the key, and Bucket, in this case Dict, is a copy constructor.

**Comment by kriskowal** [comment]
I would favor creating a named function expression for clarity.

File: fast-set.js (hunk 1 above)
Comment (vlad-alexandru-ionescu) [inline fast-set.js]: Actually, instead of passing `this` / `self` I could just use `FastSet.prototype.Bucket`. Would that be acceptable?

**Comment by vlad-alexandru-ionescu** [comment]
Updated. How's now?

**Comment by kriskowal** [comment]
I would favor self.Bucket so that it can be overridden by descendants.

**Comment by vlad-alexandru-ionescu** [comment]
Done.

**Comment by kriskowal** [comment]
Thanks. I will rebase this into a single commit locally.

**Comment by rayshan** [comment]
Thanks @vlad-alexandru-ionescu!

**Pull request closed without merging**